src/components/Checkbox/Checkbox.st.css

```diff
@@ -1,10 +1,103 @@
 :import {
-  -st-from: "wix-ui-core/index.st.css";
-  -st-named: Checkbox, CheckboxStyle;
+  -st-from: "wix-ui-core/dist/src/components/Checkbox/Checkbox.st.css";
+  -st-default: Checkbox;
+}
+
+:import {
+  -st-from: "../../palette.st.css";
+  -st-named: backgroundPrimary, backgroundSecondary, mainMutedBackground, mainMuted, main, mainHover, dangerMutedHover, danger, disabledFields, disabledCheck, dividerColor;
+}
+
+:vars {
+  medium: 16px;
 }
 
 .root {
   -st-extends: Checkbox;
-  -st-states: error, success, x-small, small;
-  -st-mixin: CheckboxStyle;
+  width: initial;
+  height: initial;
+  display: flex;
+  align-items: center;
+  outline: 0;
+}
+
+.root::box {
+  background: value(backgroundSecondary);
+  border: value(mainMuted);
+  border-radius: 4px;
+  border: solid 1px value(mainHover);
+  box-sizing: border-box;
+  width: value(medium);
+  height: value(medium);
+}
+
+.tickIcon {
+  width: 9px;
+  padding-left: 2.5px;
+  height: 9px;
+}
+
+.indeterminateIcon {
+  width: 8px;
+  padding-left: 3px;
+  height: 8px;
+}
+
+.root::childContainer {
+  font-size: 16px;
+  padding-left: 12px;
+}
+
+.root::box:active {
+  background-color: value(mainHover);
+}
+
+.root:focus-visible::box {
+  border-radius: 4px;
+  box-shadow: 0 0 0 3px #aadbfc;
+  border: solid 1px #3899ec;
```

Comment:
Set only `border-color`, because the width should be inherited from the normal state.